assets/index.html

```diff
@@ -1,71 +1,405 @@
+            pc.addTransceiver("video", { direction: 'sendonly' });
+            pc.addTransceiver("audio", { direction: 'sendonly' });
+            return pc.createOffer()
+                .then((desc) => {
+                const sdp = desc.sdp.toLowerCase();
+    
+                for (const codec of ['av1/90000', 'vp9/90000', 'vp8/90000', 'h264/90000']) {
+                    if (sdp.includes(codec)) {
+                        const opt = document.createElement('option');
+                        opt.value = codec;
+                        opt.text = codec.split('/')[0].toUpperCase();
+                        document.getElementById('video_codec').appendChild(opt);
+                    }
+                }
+    
+                for (const codec of ['opus/48000', 'g722/8000', 'pcmu/8000', 'pcma/8000']) {
+                    if (sdp.includes(codec)) {
+                        const opt = document.createElement('option');
+                        opt.value = codec;
+                        opt.text = codec.split('/')[0].toUpperCase();
+                        document.getElementById('audio_codec').appendChild(opt);
+                    }
+                }
+    
+                pc.close();
+            });
+    };
+        // initialize
+        const initialize = () => {
+            if (navigator.mediaDevices === undefined) {
+                state = ERROR;
+                errorMessage = 'can\'t access webcams or microphones. Make sure that WebRTC encryption is enabled.';
+                render();
+                return;
+            }
+    
+            navigator.mediaDevices.getUserMedia({ video: true, audio: true })
+                .then(() => Promise.all([
+                    populateDevices(),
+                    populateCodecs(),
+                ]))
+                .then(() => {
+                    state = DEVICE;
+                    render();
+                })
+                .catch((err) => {
+                    state = ERROR;
+                    errorMessage = err.toString();
+                    render();
+                });
+        };
+    
+        async function startWhip() {
+    
+            const resource = document.getElementById("resource").value;
+            if (!resource) {
+                alert("input resource")
+                return
+            }
+            
+            //get svc_level
+            const rid_value = document.getElementById("svc_level").value;
+            if (rid_value == 'a') {
+                const sendEncodings = [
+                    { rid: 'a', scaleResolutionDownBy: 2.0 }    
+                ]
+            }else if (rid_value == 'b') {
+                const sendEncodings = [
+                    { rid: 'a', scaleResolutionDownBy: 2.0 }    
+                ]
+            }else{
+                const sendEncodings = [
+                    { rid: 'c' }    
+                ]
+            }
+            const stream = onPublish();
```

Comment:
This function `onPublish` is not return any value

The stream is `undefined`

after:

<img width="473" alt="Screenshot 2023-10-18 at 23 35 10" src="https://github.com/binbat/live777/assets/18189138/d3711949-3b0f-4bfb-b68d-750805a2ba76">